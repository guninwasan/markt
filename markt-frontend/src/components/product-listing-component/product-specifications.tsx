import {
  ProductSpecsContainer,
  SpecGrid,
  SpecCategory,
  SpecItem,
} from "./product-listing-component.styles";

const ProductSpecs = ({ specs }: any) => {
  return (
    <ProductSpecsContainer>
      <h2>Product Specifications</h2>
      <SpecGrid>
        <SpecCategory>
          <h3>Basic Information</h3>
          <SpecItem>Condition: {specs.basicInfo.condition}</SpecItem>
          <SpecItem>Brand: {specs.basicInfo.brand}</SpecItem>
          <SpecItem>Model: {specs.basicInfo.model}</SpecItem>
          <SpecItem>Year: {specs.basicInfo.yearOfManufacture}</SpecItem>
        </SpecCategory>
        <SpecCategory>
          <h3>Appearance</h3>
          <SpecItem>Color: {specs.appearance.color}</SpecItem>
          <SpecItem>Dimensions: {specs.appearance.dimensions}</SpecItem>
          <SpecItem>Weight: {specs.appearance.weight}</SpecItem>
          <SpecItem>Material: {specs.appearance.material}</SpecItem>
        </SpecCategory>
        <SpecCategory>
          <h3>Performance</h3>
          <SpecItem>Battery Life: {specs.performance.batteryLife}</SpecItem>
          <SpecItem>
            Storage Capacity: {specs.performance.storageCapacity}
          </SpecItem>
          <SpecItem>Features: {specs.performance.additionalFeatures}</SpecItem>
        </SpecCategory>
        <SpecCategory>
          <h3>Warranty</h3>
          <SpecItem>{specs.warranty}</SpecItem>
        </SpecCategory>
      </SpecGrid>
    </ProductSpecsContainer>
  );
};

export { ProductSpecs };
