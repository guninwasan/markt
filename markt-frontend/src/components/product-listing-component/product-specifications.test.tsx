import React from "react";
import { render, screen } from "@testing-library/react";
import { ProductSpecs } from "./product-specifications";

const mockSpecs = {
  additional_details: "Some additional details",
  battery_life: "10 hours",
  brand: "BrandName",
  color: "Red",
  dimensions: "10x10x10 cm",
  material: "Plastic",
  model: "Model123",
  quantity: 5,
  storage_capacity: "256GB",
  weight: "1kg",
};

describe("ProductSpecifications", () => {
  it("should render the Product Specifications title", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Product Specifications")).toBeInTheDocument();
  });

  it("should render additional details", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Some additional details")).toBeInTheDocument();
  });

  it("should render battery life", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Battery Life: 10 hours")).toBeInTheDocument();
  });

  it("should render brand", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Brand: BrandName")).toBeInTheDocument();
  });

  it("should render color", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Color: Red")).toBeInTheDocument();
  });

  it("should render dimensions", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Dimensions: 10x10x10 cm")).toBeInTheDocument();
  });

  it("should render material", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Material: Plastic")).toBeInTheDocument();
  });

  it("should render model", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Model: Model123")).toBeInTheDocument();
  });

  it("should render quantity", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Quantity: 5")).toBeInTheDocument();
  });

  it("should render storage capacity", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Storage Capacity: 256GB")).toBeInTheDocument();
  });

  it("should render weight", () => {
    render(<ProductSpecs specs={mockSpecs} />);
    expect(screen.getByText("Weight: 1kg")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<ProductSpecs specs={mockSpecs} />);
    expect(container).toMatchSnapshot();
  });
});
