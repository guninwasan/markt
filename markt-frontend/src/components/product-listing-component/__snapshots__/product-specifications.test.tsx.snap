// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSpecs should match snapshot 1`] = `
<div>
  <div
    class="sc-khLCKb cYlhzu"
  >
    <h2>
      Product Specifications
    </h2>
    <div
      class="sc-cEzcPc ezsGWm"
    >
      <div
        class="sc-ovuCP bpKGpA"
      >
        <h3>
          Basic Information
        </h3>
        <p
          class="sc-eOzmre WmkIh"
        >
          Condition: 
          New
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Brand: 
          BrandName
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Model: 
          ModelX
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Year: 
          2022
        </p>
      </div>
      <div
        class="sc-ovuCP bpKGpA"
      >
        <h3>
          Appearance
        </h3>
        <p
          class="sc-eOzmre WmkIh"
        >
          Color: 
          Black
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Dimensions: 
          10x5x2 inches
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Weight: 
          1.5 lbs
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Material: 
          Plastic
        </p>
      </div>
      <div
        class="sc-ovuCP bpKGpA"
      >
        <h3>
          Performance
        </h3>
        <p
          class="sc-eOzmre WmkIh"
        >
          Battery Life: 
          10 hours
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Storage Capacity: 
          256GB
        </p>
        <p
          class="sc-eOzmre WmkIh"
        >
          Features: 
          Waterproof, Bluetooth
        </p>
      </div>
      <div
        class="sc-ovuCP bpKGpA"
      >
        <h3>
          Warranty
        </h3>
        <p
          class="sc-eOzmre WmkIh"
        >
          2 years
        </p>
      </div>
    </div>
  </div>
</div>
`;
